src/routes/notifications/notifications.controller.spec.ts

```diff
@@ -0,0 +1,243 @@
+import { range, sample } from 'lodash';
+import * as request from 'supertest';
+import {
+  fakeConfigurationService,
+  TestConfigurationModule,
+} from '../../config/__tests__/test.configuration.module';
+import { TestCacheModule } from '../../datasources/cache/__tests__/test.cache.module';
+import { NetworkResponseError } from '../../datasources/network/entities/network.error.entity';
+import {
+  mockNetworkService,
+  TestNetworkModule,
+} from '../../datasources/network/__tests__/test.network.module';
+import { DomainModule } from '../../domain.module';
+import { Chain } from '../../domain/chains/entities/chain.entity';
+import { chainBuilder } from '../../domain/chains/entities/__tests__/chain.builder';
+import { DataSourceErrorFilter } from '../common/filters/data-source-error.filter';
+import { SafeRegistration } from './entities/safe-registration.entity';
+import { registerDeviceDtoBuilder } from './entities/__tests__/register-device.dto.builder';
+import { safeRegistrationBuilder } from './entities/__tests__/safe-registration.builder';
+import { NotificationsModule } from './notifications.module';
+
+describe('Notifications Controller (Unit)', () => {
+  let app: INestApplication;
+
+  const safeConfigUrl = faker.internet.url();
+
+  beforeAll(async () => {
+    fakeConfigurationService.set('safeConfig.baseUri', safeConfigUrl);
+    fakeConfigurationService.set('exchange.baseUri', faker.internet.url());
+    fakeConfigurationService.set(
+      'exchange.apiKey',
+      faker.random.alphaNumeric(),
+    );
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        // feature
+        NotificationsModule,
+        // common
+        DomainModule,
+        TestCacheModule,
+        TestConfigurationModule,
+        TestNetworkModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalFilters(new DataSourceErrorFilter());
+
+    await app.init();
+  });
+
+  describe('POST /register/notifications', () => {
+    it('Success', async () => {
+      const chains: Chain[] = range(4).map((idx) =>
+        chainBuilder().with('chainId', idx.toString()).build(),
+      );
+      const safeRegistrations: SafeRegistration[] = chains.map((chain) =>
+        safeRegistrationBuilder().with('chain_id', chain.chainId).build(),
+      );
+      const registerDeviceDto = registerDeviceDtoBuilder()
+        .with('safeRegistration', safeRegistrations)
+        .build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrlPattern = `${safeConfigUrl}/api/v1/chains/`;
+        if (url.includes(getChainUrlPattern)) {
+          return Promise.resolve({ data: sample(chains) });
+        }
+        return Promise.reject(`No matching rule for url: ${url}`);
+      });
+      mockNetworkService.post.mockImplementation(() => Promise.resolve());
```

Comment:
We should match against a URL so that we don't consider "any" `post` call.

Reply:
In this case we also have to check the URL matches against any of the `transactionService` properties of the `chains`. Something like:
```javascript
mockNetworkService.post.mockImplementation((url) => {
  if (
    chains.some(
      (chain) =>
        url ===
        `${chain.transactionService}/api/v1/notifications/devices/`,
    )
  ) {
    return Promise.resolve();
  }
  return Promise.reject(`No matching rule for url: ${url}`);
});
```

Please ping me if you see a simpler solution 🙂

Reply:
https://github.com/5afe/safe-client-gateway-nest/pull/267/commits/0728521aa558fa73c90498d9c13798e313936c93